pkg/controller/secretsmanager/controller.go

```diff
@@ -0,0 +1,287 @@
+			return managed.ExternalObservation{}, errors.New(errK8sSecretNotFound)
+		}
+		resourceUpToDate = secretsmanager.IsUpToDate(cr, req, secret, rsp)
+	}
+	if req.DeletedDate != nil {
+		resourceUpToDate = true
+		cr.Status.SetConditions(runtimev1.Deleting().WithMessage(secretMarkedForDeletion))
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:          true,
+		ResourceUpToDate:        resourceUpToDate,
+		ResourceLateInitialized: !cmp.Equal(current, &cr.Spec.ForProvider),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.Secret)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotSecret)
+	}
+
+	cr.SetConditions(runtimev1.Creating())
+
+	secret, err := secretsmanager.GetSecret(ctx, e.kube, cr)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+
+	_, err = e.client.CreateSecretRequest(secretsmanager.GenerateCreateSecretsmanagerInput(meta.GetExternalName(cr), &cr.Spec.ForProvider, secret)).Send(ctx)
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreateFailed)
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+
+	cr, ok := mg.(*v1alpha1.Secret)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotSecret)
+	}
+
+	req, err := e.client.DescribeSecretRequest(&awssecretsmanager.DescribeSecretInput{SecretId: awscommon.String(meta.GetExternalName(cr))}).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(secretsmanager.IsErrorNotFound, err), errDescribeSecretFailed)
+	}
+
+	err = e.updateTags(ctx, cr, req)
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	secret, err := secretsmanager.GetSecret(ctx, e.kube, cr)
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	_, err = e.client.UpdateSecretRequest(secretsmanager.GenerateUpdateSecretInput(meta.GetExternalName(cr), cr.Spec.ForProvider, secret)).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(secretsmanager.IsErrorNotFound, err), errUpdateFailed)
+	}
+
+	if cmp.Equal(v1alpha1.GenerateSecretTags(cr.Spec.ForProvider.Tags), req.Tags) {
+		_, err = e.client.TagResourceRequest(&awssecretsmanager.TagResourceInput{
+			SecretId: awscommon.String(meta.GetExternalName(cr)),
+			Tags:     v1alpha1.GenerateSecretTags(cr.Spec.ForProvider.Tags),
+		}).Send(ctx)
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(secretsmanager.IsErrorNotFound, err), errTagResourceRequest)
+		}
+	}
+
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
+
+	cr, ok := mg.(*v1alpha1.Secret)
+	if !ok {
+		return errors.New(errNotSecret)
+	}
+
+	cr.SetConditions(runtimev1.Deleting())
+
+	if cr.Status.AtProvider.DeletionDate != nil {
+		// only request a new deletion if the user has changed the recovery window
```

Comment:
I couldn't get this part. Why do we need to do this calculation?

Reply:
This calculation was necessary, because we combined `ForceDeleteWithoutRecovery` and `RecoveryWindowInDays`. I split that up again, so this code isn't needed anymore.

Reply:
As mentioned below `DeletionDate` is set based on the response of the `DeleteSecretRequest`. After it is set once, we check whether the `RecoveryWindowInDays` has changed as the resource in Kubernetes is only deleted after the grace period is over. If it has not changed, there is no need to call `DeleteSecretRequest` again and update the `DeletionDate`.

Reply:
Another comment from my end: We had a look what would happen if you don't add a `RecoveryWindowInDays` and do not add a `ForceDeleteWithoutRecovery`. In that case, the secret is still deleted and the `DeletionDate` is set to 30 days. `RecoveryWindowInDays` is therefore implicit.